orchestrator/cli/domain_gen_helpers/resource_type_helpers.py

```diff
@@ -0,0 +1,341 @@
+                "Resource type description: "
+            )
+
+        return sql_compile(
+            Insert(ResourceTypeTable).values({"resource_type": resource_type, "description": description})
+        )
+
+    return [create_resource_type(resource_type) for resource_type in resource_types]
+
+
+def generate_update_resource_types_sql(resource_types: Dict[str, str]) -> List[str]:
+    """Generate SQL to update resource types.
+
+    Args:
+        - resource_types: Dict with new resource type name by old resource type name.
+            - key: old resource type name.
+            - value: new resource type name.
+
+    Returns: List of SQL strings to update resource types.
+    """
+
+    def update_resource_type(old_rt_name: str, new_rt_name: str) -> str:
+        return sql_compile(
+            Update(ResourceTypeTable)
+            .where(ResourceTypeTable.resource_type == old_rt_name)
+            .values(resource_type=new_rt_name)
+        )
+
+    return [update_resource_type(*item) for item in resource_types.items()]
+
+
+def generate_delete_resource_types_sql(resource_types: List[str]) -> List[str]:
+    """Generate SQL to delete resource types.
+
+    Args:
+        - resource_types: List of resource type names.
+
+    Returns: List of SQL strings to delete resource types.
+    """
+    if not resource_types:
+        return []
+    return [
+        sql_compile(
+            Delete(SubscriptionInstanceValueTable).where(
+                SubscriptionInstanceValueTable.resource_type_id.in_(get_resource_types(resource_types))
+            )
+        ),
+        sql_compile(Delete(ResourceTypeTable).where(ResourceTypeTable.resource_type.in_(resource_types))),
+    ]
+
+
+def generate_create_resource_type_relations_sql(resource_types: Dict[str, Set[str]]) -> List[str]:
+    """Generate SQL to create resource type relations.
+
+    Args:
+        - resource_types: Dict with product blocks by resource type
+            - key: Resource type name.
+            - value: Set of product block names to relate with.
+
+    Returns: List of SQL strings to create relation between product blocks and resource type.
+    """
+
+    def create_resource_type_relation(resource_type: str, block_names: Set[str]) -> str:
+        resource_type_id_sql = get_resource_type(resource_type)
+
+        def create_block_relation_dict(block_name: str) -> Dict[str, Union[str, Query]]:
+            block_id_sql = get_product_block_id(block_name)
+            return {"resource_type_id": resource_type_id_sql, "product_block_id": block_id_sql}
+
+        block_relation_dicts = [create_block_relation_dict(block_name) for block_name in block_names]
+        return sql_compile(Insert(product_block_resource_type_association).values(block_relation_dicts))
+
+    return [create_resource_type_relation(*item) for item in resource_types.items()]
+
+
+def generate_create_resource_type_instance_relations_sql(
```

Comment:
Nitpick: this function is called `generate_create_resource_type_instance_relations_sql` but it returns resource type instance values